Unicode/charset/iso_8859_16.c

```diff
@@ -0,0 +1,176 @@
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  /*** 0xc0 ***/
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+static const unsigned char i8859_16_from_unicode_20[256] = {
+  /*** 0x00 ***/
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xb5, 0xa5, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  /*** 0x40 ***/
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  /*** 0x80 ***/
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xa4, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  /*** 0xc0 ***/
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+static const unsigned char * const i8859_16_from_unicode_[] = {
+    i8859_16_from_unicode_0,
+    i8859_16_from_unicode_1,
+    i8859_16_from_unicode_2,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    allzeros,
+    i8859_16_from_unicode_20
+};
+
+struct charmap i8859_16_from_unicode = { 0, 32, (unsigned char **) i8859_16_from_unicode_, (unichar_t *) unicode_from_i8859_16 };
```

Comment:
This is the way these 8859-* (plus other files) are built using the existing generate*.sh script.
Very little modification done beyond adding the missing 8859-16 chart.

This would probably be okay with older computers running a smaller MAXC, but it is showing itself as a bit wasteful of program binary/memory space. I haven't changed how it's built, but looking at these maps, plus unialt.c, I think these could be made a bit more memory/program efficient if we (eventually) convert these to function calls (is_Ligatures.c is an example where we go beyond MAXC to include the full Unicode chart).

In terms of 8859-16, it exists, plus there are keyboards for it as well (wikipedia). I'm a little puzzled why it wasn't added, maybe a trade-off due to memory limits in 2001-ish, which isn't such an issue these days.